Only enable git import submit button when a branch or tag is selected

eclarizio opened the pull request:
This is an addendum to the actual fix for the bug by @mkanoor. This PR will simply disable the submit button until a branch or tag has been selected, but if someone were to manually enable the button and attempt to submit without a branch or tag, that error is caught and shown to the user.

It also includes a more targeted selector for the `.git-branches` and `.git-tags` elements, as the selectpicker javascript generates a ul/li combination alongside the actual select element.

We also decided it would be best to default to the "origin/master" branch, if it's available, otherwise the first branch in the drop-down, instead of having the user pick between branch or tag at first.

Links
----------------
https://bugzilla.redhat.com/show_bug.cgi?id=1386296

syncrou commented on app/assets/javascripts/automate_import_export.js:
```diff
@@ -118,10 +118,10 @@ var Automate = {
-      $('.git-tags').find('option').remove().end();
-      $('.git-branches').selectpicker('refresh');
-      $('.git-tags').selectpicker('refresh');
+      $('select.git-branches').find('option').remove().end();
```
@eclarizio - Can we include in the description of the PR why this change was needed?

gmcculloug commented:
Looks good.  @eclarizio Please update the description per @syncrou and we can merge.

himdel commented on app/assets/javascripts/automate_import_export.js:
```diff
@@ -213,6 +214,14 @@ var Automate = {
 
   setUpGitRefreshClickHandlers: function() {
+    var toggleSubmitButton = function() {
+      if ($('.git-branch-or-tag').val() !== "") {
```
This should probably be `$('.git-import-submit').prop('disabled', $('.git-branch-or-tag').val() !== "");

eclarizio replied:
How about
```
var branchOrTagSelected = $('.git-branch-or-tag').val() === "";
$('.git-import-submit').prop('disabled', branchOrTagSelected);
```

himdel replied:
Even better :)

eclarizio commented:
@himdel Ok, refactored that part.

@syncrou Added a bit more to the description, also added some logic for selecting "origin/master" or the first available branch.

@mkanoor I think this covers what we were talking about for the branch selection, let me know if you had something else in mind.

gmcculloug commented:
Tested this change and it's looking good.  Test failures seem unrelated.

@syncrou @mkanoor Please review.

mkanoor commented:
@gmcculloug  tested looks good

chessbyte commented:
Euwe BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1397413

chessbyte commented:
Euwe Backport details:
```sh
$ git log -1
commit d438e4cc51f85d5524b149161250bcc87d271340
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Mon Nov 21 19:50:04 2016 -0500

    Merge pull request #12753 from eclarizio/BZ1386296
    
    Only enable git import submit button when a branch or tag is selected
    (cherry picked from commit c39a7200ad4402f3929488cef9ce6aa530c2f54e)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1397413
```
